lib/streamlit/type_util.py

```diff
@@ -455,6 +457,9 @@ def convert_anything_to_df(
     if is_type(df, "numpy.ndarray") and len(df.shape) == 0:
         return pd.DataFrame([])
 
+    if is_type(df, _PYSPARK_DF_TYPE_STR) and callable(getattr(df, "toPandas")):
+        df = df.toPandas()
```

Comment:
Aren't Pyspark dataframes also uneveluated? Maybe we should apply the limit here as well?

Reply:
Good point 👍 It's updated now and we're limiting the number of rows the same way, we're limiting them for Snowpark 👍